src/openApi/v2/parser/getOperationName.ts

```diff
@@ -7,14 +7,19 @@ import sanitizeOperationName from '../../../utils/sanitizeOperationName';
  * This will use the operation ID - if available - and otherwise fallback
  * on a generated name from the URL
  */
-export const getOperationName = (url: string, method: string, operationId?: string): string => {
-    if (operationId) {
+export const getOperationName = (
+    url: string,
+    method: string,
+    useOperationId: boolean,
+    operationId?: string
+): string => {
+    if (useOperationId && operationId) {
         return camelCase(sanitizeOperationName(operationId).trim());
     }
 
     const urlWithoutPlaceholders = url
         .replace(/[^/]*?{api-version}.*?\//g, '')
-        .replace(/{(.*?)}/g, '')
+        .replace(/{(.*?)}/g, 'by-$1')
```

Comment:
@askvortcov can you explain why this change is necessary?

Reply:
@mrlubos That is to handle the case of:
GET /projects/
GET /projects/{id}

The name generated for the second line becomes `getProjects_1` which I find useless and unreadable. 

With the change, the name becomes `getProjectsById` which IMO represents the intent way better.

Please see the test for examples.

Reply:
Thanks, that works. I think that might fix https://github.com/ferdikoomen/openapi-typescript-codegen/issues/1952, but cautious of breaking any codebases depending on the current behaviour